compiler/src/yul/mappers/expressions.rs

```diff
@@ -105,6 +112,20 @@ fn expr_call(context: &Context, exp: &Spanned<fe::Expr>) -> Result<yul::Expressi
                 .collect::<Result<_, _>>()?;
 
             return match call_type {
+                CallType::BuiltinFunction { func } => match func {
+                    GlobalMethod::Keccak256 => {
+                        let first_arg = args.node.first().expect("Missing argument");
+                        let arg_expr = context
+                            .get_expression(first_arg)
+                            .expect("invalid attributes");
+                        let size = to_fixed_size(&arg_expr.typ)?;
+                        let func_name: &str = func.into();
+
+                        let func_name = identifier! { (func_name) };
+                        let size = identifier_expression! { (size.size()) };
+                        Ok(expression! { [func_name]([yul_args[0].to_owned()], [size]) })
```

Comment:
https://docs.soliditylang.org/en/latest/yul.html#simple-example

IIUC `keccak256(p, n)` is expecting a pointer to memory (`keccak(mem[p…(p+n)))`) so calling it with a stack value like `keccak256(1)` doesn't do what it is supposed to. It would return the keccak over the 32 bytes starting at memory address `1` which is absolutely unrelated to the value `1`.

I think what we should probably do is to validate that thergument to `keccak256` is always of location `Memory`. So valid examples would be `keccak256(mem_var)` or `keccak256(storage_var.to_mem())` and probably `keccak256(abi_encode_packed(1))` (because the latter would turn `1` into a memory pointer where the value is encoded.